src/app/games/abalone/AbaloneRules.ts

```diff
@@ -64,22 +64,23 @@ export class AbaloneRules extends Rules<AbaloneMove, AbaloneGameState, AbaloneLe
     }
     private static isLegalRealPush(firstEnnemy: Coord,
                                    move: AbaloneMove,
-                                   state: AbaloneGameState,
+                                   state: AbaloneState,
                                    pushingPieces: number,
-                                   newBoard: number[][])
+                                   newBoard: FourStatePiece[][])
     : AbaloneLegalityStatus
     {
         let enemyPieces: number = 0;
-        const ENEMY: number = state.getCurrentEnnemy().value;
-        const PLAYER: number = state.getCurrentPlayer().value;
+        const ENEMY: FourStatePiece = FourStatePiece.ofPlayer(state.getCurrentEnnemy());
+        const PLAYER: FourStatePiece = FourStatePiece.ofPlayer(state.getCurrentPlayer());
+        // TODO: find a way for Player not to be assigned to FourStatePiece putain
```

Comment:
Clue about that ?

Reply:
est-ce que l'idéal ça serait pas d'avoir `value` en privé? Ainsi pas de soucis de typage structurel non plus où un FourStatePiece et Player sont compatibles

Reply:
l'intérêt de l'encapsulation ici c'est de laisser libre l'implémentation interne: si on veut passer par une enum ou autre, on est certain que personne ne fait de .value. L'intérêt de rendre privé c'est d'éviter que quelqu'un fasse le crado et aie du .value partout justement

Reply:
On utilise encore le .value pour tabuler içi mais oui, je vois. Je serais plus chaud de garder les choses comme ça mais au pire ça peut se ticketter plus tard.